### PR TITLE
Fix syntax error.

### PR DESCRIPTION
**Pull request by mlimber**
Problem:
-

Solution:
-

Issue: #

Reviewers:
@

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -258,9 +258,14 @@ struct Streamable {
...
          class Engine   = Streamable(class EngineStream)>
 class example {
+public:
   example(Exchange exchange, Engine engine)
-    : exchange(exchange), engine(engine)
+    : exchange(std::move(exchange)), engine(std::move(engine))
   { }
+  
+private:
+  Streamable exchange;
```
Comment (kris-jusiak) [inline README.md]: Right, that should be
```
Exchange exchange;
Engine engine;
```

File: README.md [hunk 2]
```diff
@@ -103,7 +103,7 @@ retq
 
 ```cpp
 struct interface {
-  virtual ~iworld() noexcept = default;
+  virtual ~interface() noexcept = default;
```
Comment (kris-jusiak) [inline README.md:107]: Good catch @mlimber.
Would you be able to add a link to a working example -> https://wandbox.org/permlink/so4nFDduIzx7m5sV
Thanks!

Comment (kris-jusiak) [inline README.md:107]: There is the same issue with bind interface :/ (it should be ~interface)
Working link -> https://wandbox.org/permlink/4yBTlH1jvdrsRPes

File: README.md (hunk 1 above)
Comment (kris-jusiak) [reply README.md]: Link to a working example -> https://wandbox.org/permlink/9QZTRsLkPNholmpj

File: README.md [hunk 3]
```diff
@@ -258,9 +258,14 @@ struct Streamable {
...
 class example {
+public:
   example(Exchange exchange, Engine engine)
-    : exchange(exchange), engine(engine)
+    : exchange(std::move(exchange)), engine(std::move(engine))
   { }
+  
+private:
+  Streamable exchange;
+  Streamable engine;
```
Comment (kris-jusiak) [inline README.md]: There are also a few issues with `bind templates example` :/
Link to a working example here -> https://wandbox.org/permlink/wqrF7gZGOM7Jusx9

**Comment by kris-jusiak** [comment]
Thanks, @mlimber.

If you could please fix other examples and add links to wandbox to the README that would be really appreciated.
I would also recommend rebasing/squashing your commits into one for better history.

Thanks.

File: README.md (hunk 2 above)
Comment (mlimber) [reply README.md:107]: I'm not sure how to fix the template param binding: https://wandbox.org/permlink/ja8x7xXpl5OQMH0b

I can get it to work if the type is not used for inheritance. Not sure what's going on.

Comment (kris-jusiak) [reply README.md:107]: https://wandbox.org/permlink/wqrF7gZGOM7Jusx9

**Comment by mlimber** [comment]
Closed in lieu of https://github.com/boost-experimental/di/pull/387

**Comment by mlimber** [comment]
Closed in lieu of https://github.com/boost-experimental/di/pull/387.

**Pull request closed without merging**